packages/test-base/src/commonTest/kotlin/io/realm/kotlin/test/common/RealmResultsTests.kt

```diff
@@ -68,6 +69,21 @@ class RealmResultsTests {
         assertEquals(1, realm.query<Parent>("name CONTAINS '1'").find().query("name CONTAINS '2'").count().find())
     }
 
+    @Test
+    fun query_returnBackingQuery() {
+        val query = realm.query<Parent>("name CONTAINS '1'")
+        val backingQuery = query.find().query("")
```

Comment:
Should this really be allowed in the public API?

Reply:
I was thinking about that, but ended up on it probably being okay, this is basically similar to query.find().query("TRUEPREDICATE"), except you don't end up with `<originalQuery> AND TRUEPREDICATE`

We could also expose a new method `RealmResults.query` that would return the backing query, but it didn't seem worth it to expand the API surface? But not sure what you think?

I did miss updating the `RealmResults.query` though, this has been fixed.

Reply:
Hmm. Not feeling strongly against it and I see your point about API surface, but ... Would it be a compromise to do a default `query: String = TRUE_PREDICATE` argument so it matched the `TypedRealm.query` method and then match that internally in the implementation instead of the empty string.

Reply:
Yes, that sounds fair 👍